pkg/model/stack_expander.go

```diff
@@ -21,9 +57,8 @@ func expandEnvScalarNode(node *yaml3.Node) (*yaml3.Node, error) {
 	// when is a Sequence and starts with $ can be a list of envs, so transform the list to key=value format
 	case yaml3.SequenceNode:
 		for indx, subNode := range node.Content {
-			if strings.HasPrefix(subNode.Value, "$") {
+			if key, ok := isEnvStringKey(subNode.Value); ok {
```

Comment:
@teresaromero I think we don't need logic for this, it's automatically done by the golang env substitution (cc: @jLopezbarb)

Reply:
i added this because there is the case where subNode.Value can be $ENV or $ENV:src, this is used to append as key de variable name, after the env is expanded, with the prev logic `ENV:src` was used as key, which is not correct

Reply:
@pchico83 added unit test for the functions so its clearer the changes and what is doing the functions

cc: @ifbyol

Reply:
does `${ENV}:src` work today?

Reply:
> does `${ENV}:src` work today?

i'm afraid not, this is an example of the issue reported... 
this serialiser was implemented tu support multiline envs in the first place (PR description). Are we using a compose serializer? I believe that whenever we are changing to use the compose serializer this would need to be checked